app/Livewire/Components/Pages/Gallery/Album.php

```diff
@@ -203,19 +204,22 @@ private function fetchHeaderUrl(): SizeVariant|null
 			return $headerSizeVariant;
 		}
 
-		$photo = SizeVariant::query()
+		$query_ratio = SizeVariant::query()
 					->select('photo_id')
 					->whereBelongsTo($this->album->photos)
 					->where('ratio', '>', 1) // ! we prefer landscape first.
-					->whereIn('type', [SizeVariantType::MEDIUM, SizeVariantType::SMALL2X, SizeVariantType::SMALL])
-					->inRandomOrder()
-					->first() ??
-				SizeVariant::query()
-					->select('photo_id')
-					->whereBelongsTo($this->album->photos)
-					->whereIn('type', [SizeVariantType::MEDIUM, SizeVariantType::SMALL2X, SizeVariantType::SMALL])
-					->inRandomOrder()
-					->first();
+					->whereIn('type', [SizeVariantType::MEDIUM, SizeVariantType::SMALL2X, SizeVariantType::SMALL]);
+		$num = $query_ratio->count() - 1;
+		$photo = $query_ratio->skip(rand(0, $num))->first();
+
+		if ($photo === null) {
```

Comment:
Would this test work with `$num === 0` instead? That might be harder to read though.